dudect/src/main.rs

```diff
@@ -0,0 +1,52 @@
+//! Fuzz tests for `crypto-bigint`.
+//!
+//! These use the `dudect_bencher` crate to check certain operations for
+//! constant-time behavior.
+
+use crypto_bigint::{Limb, UInt, U256};
+use dudect_bencher::{ctbench_main, ctbench_main_with_seeds, BenchRng, Class, CtRunner};
+use rand::{Rng, RngCore};
+
+/// Generate a random `UInt`.
+// TODO(tarcieri): use `UInt::random` when `dudect_bencher` updates `rand`
+fn rand_uint<const LIMBS: usize>(rng: &mut BenchRng) -> UInt<LIMBS> {
+    let mut limbs = [Limb::ZERO; LIMBS];
+
+    for limb in &mut limbs {
+        *limb = if cfg!(target_pointer_width = "32") {
+            rng.next_u32().into()
+        } else if cfg!(target_pointer_width = "64") {
+            rng.next_u64().into()
+        } else {
+            unimplemented!("only 32-bit and 64-bit targets are supported")
+        }
+    }
+
+    limbs.into()
+}
+
+/// Check `UInt::reduce` for constant-time operation.
+fn reduce(runner: &mut CtRunner, rng: &mut BenchRng) {
+    let mut inputs = Vec::<U256>::new();
+    let mut classes = Vec::new();
+
+    // Make 100,000 inputs on each run
+    for _ in 0..100_000 {
+        inputs.push(rand_uint(rng));
+
+        // Randomly pick which distribution this example belongs to
+        if rng.gen::<bool>() {
+            classes.push(Class::Left);
+        } else {
+            classes.push(Class::Right);
+        }
```

Comment:
@rozbb is this the best way to do this?

Or should I perhaps make one of the two `Class`es always the same?